core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/common/Counter.java

```diff
@@ -0,0 +1,30 @@
+package org.optaplanner.constraint.streams.bavet.common;
+
+import static org.optaplanner.constraint.streams.bavet.common.BavetTupleState.DEAD;
+
+import java.util.function.IntPredicate;
+
+public final class Counter<Tuple_ extends Tuple> {
```

Comment:
Naming wise, this name is too broad if it's a top level class.

Proposal A: ExistsCounter